components/charts/AdvancedTimeline.tsx

```diff
@@ -251,30 +251,33 @@ export default class AdvancedTimeline extends React.PureComponent<Props, State>
   };
 
   updateTooltipPos = (xPos: number) => {
-    const firstSerie = this.props.series[0];
-    if (this.state.mouseOver && firstSerie) {
-      const { updateTooltip } = this.props;
-      const date = this.state.xScale.invert(xPos);
-      const bisectX = bisector<T.Chart.Point, Date>(d => d.x).right;
-      let idx = bisectX(firstSerie.data, date);
-      if (idx >= 0) {
-        const previousPoint = firstSerie.data[idx - 1];
-        const nextPoint = firstSerie.data[idx];
-        if (
-          !nextPoint ||
-          (previousPoint &&
-            date.valueOf() - previousPoint.x.valueOf() <= nextPoint.x.valueOf() - date.valueOf())
-        ) {
-          idx--;
-        }
-        const selectedDate = firstSerie.data[idx].x;
-        const xPos = this.state.xScale(selectedDate);
-        this.setState({ selectedDate, selectedDateXPos: xPos, selectedDateIdx: idx });
-        if (updateTooltip) {
-          updateTooltip(selectedDate, xPos, idx);
+    this.setState(state => {
```

Comment:
NP: you probably could continue to avoid calling the `setState` if conditions (L256) are not fulfilled. The `return null` at the end of the `setState` is weird IMO.

Reply:
The `return null` is used because we don't want to update the state, I think we could also return an empty object and it would do the same.

I can't move the full condition of (L256) outside the `setState` since it's based on the state. But ok I could split the condition to avoid calling the `setState` if the `firstSerie` is `undefined`, but then it add one more layer of `if` around everything and just make things a bit more complex for probably not much performance gain (just avoiding a function call basically)...